test/dotnet-new3.UnitTests/DotnetNewSearch.cs

```diff
@@ -877,5 +885,36 @@ public int Compare(string? left, string? right)
                 return string.Compare(left, right, StringComparison.OrdinalIgnoreCase);
             }
         }
+
+        private class DownloadCountComparer : IComparer<string>
```

Comment:
When we change the implementation in CLI project in the future, will there be some kind of reminder to tell us that we need to change this class as well?

Reply:
Then it would be a valid test failure :)

Reply:
But then this is only testing if the code in CLI project is the same as the one in `dotnet-new3.UnitTests` project. It doesn't test if column `12300k` comes before `<1k`. If there is a bug in this comparer and a package with unknown count is always at the top, the test won't be able to catch it. Right?

If that is the case, then maybe we can fix this by writing another test method to test just this class in UnitTests project. 

Or maybe I'm just having its-friday-evening-maybe-you-should-stop-working-for-the-week syndrome, not sure 🙂

Reply:
well, my thinking is as follows:
- if this part is accidentally broken, this piece of code will allow to catch it
- if this part is intentionally changed, this piece of code should also change. 

At the moment, it's not exactly the same as in CLI, and I cannot think of any other approach to verify sorting, do you?

Reply:
We can verify the sorting by writing a new test method that tests this `DowlnoadCountComparer.Compare` method. This would simply look something like this:
```csharp
[DataRow("12k", "3k", 1)]
[DataRow(null, "", -1)]
[DataRow("<1k", null, 1)]
[DataRow("7k", "7k", 0)]
[DataRow("<1k", "<1k", 0)]
public void TestCompare(string? x, string?y, int expectedOutcome)
{
    Assert.AreEqual(expectedOutcome, new DownloadCountComparer().Compare(x, y));
}
```

Reply:
I added the tests for TotalDownloadsComparer (that now uses long). 
I don't think comparer in tests :) need a test.